File: cSharp/Api-ng-sample-code/Api-ng-sample-code/TO/MarketBettingType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Api_ng_sample_code.TO
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum MarketBettingType
    {
        ODDS,
        LINE,
        RANGE,
        ASIAN_HANDICAP_DOUBLE_LINE,
        ASIAN_HANDICAP_SINGLE_LINE,
        FIXED_ODDS
    }
}

File: java/ng/src/main/java/com/betfair/aping/enums/MarketBettingType.java
package com.betfair.aping.enums;

public enum MarketBettingType {
	ODDS("ODDS"), 
	LINE("LINE"), 
	RANGE("RANGE"), 
	ASIAN_HANDICAP_DOUBLE_LINE("ASIAN_HANDICAP_DOUBLE_LINE"),
	ASIAN_HANDICAP_SINGLE_LINE("ASIAN_HANDICAP_SINGLE_LINE"), 
	FIXED_ODDS("FIXED_ODDS");
	
	private String value;

	private MarketBettingType(String value) {
		this.value=value;
	}

	private MarketBettingType(int id) {
		value=String.format("%04d", id);
	}
	
	public String getCode() {
		return value;
	}
}
